drivers/backup/portworx/portworx.go

```diff
@@ -334,6 +334,22 @@ func (p *portworx) GetClusterName(ctx context.Context, orgID string, clusterUid
 	return "", fmt.Errorf("cluster with uid '%s' not found for org '%s'", clusterUid, orgID)
 }
 
+func (p *portworx) GetClusterStatus(orgID string, clusterName string, ctx context.Context) (api.ClusterInfo_StatusInfo_Status, error) {
+	clusterEnumerateReq := &api.ClusterEnumerateRequest{
+		OrgId: orgID,
+	}
+	enumerateRsp, err := p.EnumerateCluster(ctx, clusterEnumerateReq)
+	if err != nil {
+		return api.ClusterInfo_StatusInfo_Invalid, err
+	}
+	for _, cluster := range enumerateRsp.GetClusters() {
+		if cluster.GetName() == clusterName {
+			return cluster.Status.Status, nil
+		}
+	}
+	return api.ClusterInfo_StatusInfo_Invalid, fmt.Errorf("cluster with name '%s' not found for org '%s'", clusterName, orgID)
+}
+
```

Comment:
You can simplify the function by using a different api endpoint:

```
func (p *portworx) GetClusterStatus(orgID string, clusterName string, ctx context.Context) (api.ClusterInfo_StatusInfo_Status, error) {
	clusterInspectReq := &api.ClusterInspectRequest{
		OrgId: orgID,
		Name: clusterName,
	}
	inspectRsp, err := p.InspectCluster(ctx, clusterInspectReq)
	if err != nil {
		return api.ClusterInfo_StatusInfo_Invalid, err
	}
	return inspectRsp.Cluster.Status.Status, nil
}
```

Reply:
I just noticed while writing the simplified code for _this_ PR, that in an older PR of yours, we can further simplify: https://github.com/portworx/torpedo/pull/1195#discussion_r1142934124
Though, it's working, so it might be fine.

Reply:
To maintain consistency with the legacy codebase, I am using EnumerateCluster since many of the helper functions in the older code rely on EnumerateRequest.

Reply:
This function will perform much better if you switch to `inspectCluster`. Also, I don't see how this will affect other functions which are using the enumerate method, as I'm not suggesting the removal of anything.
I feel consistency is useful for code readability, but in this case, I see no benefit in being consistent, especially when the alternative method (inspect) is faster and smaller.

@mkoppal-px  what do you suggest?